radixdlt-java/src/main/java/com/radixdlt/client/core/network/actions/SubmitAtomRequestAction.java

```diff
@@ -22,6 +22,10 @@ private SubmitAtomRequestAction(String uuid, Atom atom) {
 	}
 
 	public static SubmitAtomRequestAction newRequest(Atom atom) {
+		if (atom.getRequiredFirstShard().isEmpty()) {
```

Comment:
Should this kind of logic be moved into a validator?

Reply:
yes, but not for now